### PR TITLE
feat: Add a configurable decimals flag to `roller config init`

### PR DESCRIPTION
**Pull request by ItayLevyOfficial**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

For Author:

- [x]  Targeted PR against correct branch
- [x]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x]  Linked to Github issue with discussion and accepted design
- [x]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

File: cmd/config/init/flags.go [hunk 1]
```diff
@@ -98,3 +107,26 @@ func verifyTokenSupply(cmd *cobra.Command) error {
...
+		return err
+	}
+	if decimals > 18 {
+		return fmt.Errorf("invalid decimals: %d. Must be less than or equal to 18", decimals)
+	}
+	return nil
+}
+
+func isValidDenom(s string) bool {
+	if len(s) != 3 {
```
Comment (mtsitrin) [inline cmd/config/init/flags.go:123]: is it standard?

File: cmd/config/init/output.go [hunk 2]
```diff
@@ -20,6 +28,24 @@ func printInitOutput(addresses []utils.AddressData, rollappId string) {
...
 		}
 	}
-	utils.PrintAddresses(filteredAddresses)
-	fmt.Printf("\n🔔 Please fund these addresses to register and run the rollapp.\n")
+	return filteredAddresses
+}
+
+func FormatTokenSupplyLine(rollappConfig utils.RollappConfig) string {
+	displayDenom := strings.ToUpper(rollappConfig.Denom[1:])
+	return fmt.Sprintf("💰 Total Token Supply: %s %s. Note that 1 %s == 1 * 10^%d %s (like 1 BTC == 1 * 10^8 satoshi).",
```
Comment (mtsitrin) [inline cmd/config/init/output.go:36]: Why not giving the ETH example? we set our decimals according to that

File: cmd/config/init/flags.go [hunk 3]
```diff
@@ -19,6 +21,10 @@ func addFlags(cmd *cobra.Command) {
 	cmd.Flags().StringP(FlagNames.HubID, "", StagingHubID, fmt.Sprintf("The ID of the Dymension hub. %s", getAvailableHubsMessage()))
 	cmd.Flags().StringP(FlagNames.RollappBinary, "", "", "The rollapp binary. Should be passed only if you built a custom rollapp")
 	cmd.Flags().StringP(FlagNames.TokenSupply, "", defaultTokenSupply, "The total token supply of the RollApp")
+	cmd.Flags().UintP(FlagNames.Decimals, "", 18,
```
Comment (mtsitrin) [inline cmd/config/init/flags.go:24]: IMO this flag should be hidden for the default case
It's just confusing, and we anyway expect it to be 18

File: cmd/config/init/init.go [hunk 4]
```diff
@@ -26,6 +28,13 @@ func InitCmd() *cobra.Command {
 			if !validateRollAppID(rollappID) {
 				return fmt.Errorf("invalid RollApp ID '%s'. %s", rollappID, getValidRollappIdMessage())
 			}
+			denom := args[1]
+			if !isValidDenom(denom) {
+				return fmt.Errorf("invalid denom '%s'. %s", denom, validDenomMsg)
```
Comment (omritoptix) [inline cmd/config/init/init.go:33]: why not just return the error with the message from the isValidDenom?

File: cmd/config/init/flags.go (hunk 1 above)
Comment (ItayLevyOfficial) [reply cmd/config/init/flags.go:123]: No, but we add a 'u' in the beginning. What would make more sense for you?

File: cmd/config/init/output.go (hunk 2 above)
Comment (ItayLevyOfficial) [reply cmd/config/init/output.go:36]: good idea

File: cmd/config/init/flags.go (hunk 1 above)
Comment (mtsitrin) [reply cmd/config/init/flags.go:123]: I meant why the SYMBOL is enforced to be 3 letters? can't it be SAMBA with uSAMBA?

Comment (ItayLevyOfficial) [reply cmd/config/init/flags.go:130]: Ye, which restriction makes more sense to you: allowing only three to five alphabetic letters? I think this enforcement makes sense

Comment (mtsitrin) [reply cmd/config/init/flags.go:86]: I think three to five is better

**Review by mtsitrin — APPROVED** [review]

**Pull request merged**